src/py/flwr/common/message.py

```diff
@@ -30,25 +32,73 @@ class Metadata:
         An identifier for the current run.
     message_id : str
         An identifier for the current message.
+    src_node_id : int
+        An identifier for the node sending this message.
+    dst_node_id : int
+        An identifier for the node receiving this message.
+    reply_to_message : str
```

Comment:
That's a valid option. I keep it as `str` to be consistent with other fields. For example, the `ttl` field cannot set to `None`; instead we treat `""` as no TTL.

Reply:
True, we should keep the approach consistent. We can do that here. Or should we make them both `Optional` to enable things like `if msg.ttl:` (instead of the more unintuitive version`if msg.ttl == "":`)?